pyaedt/modules/solutions.py

```diff
@@ -2245,153 +2066,284 @@ def scale(value=1):
-            pointa=(0.35, 0.25),
-            pointb=(0.64, 0.25),
-            style="modern",
-            event_type="always",
-        )
-        p.add_slider_widget(
-            uf.update_theta2,
-            rng=[-180, 180],
-            value=0,
-            title="Theta2",
-            pointa=(0.67, 0.25),
-            pointb=(0.98, 0.25),
-            style="modern",
-            event_type="always",
-        )
-        sargs = dict(height=0.4, vertical=True, position_x=0.05, position_y=0.5)
-        # ff_mesh_inst = p.add_mesh(uf.output,smooth_shading=True,cmap="jet",scalar_bar_args=sargs,opacity=0.5)
-        # not sure why, but smooth_shading causes this to not update
-        plot_min = self.min_gain
-        ff_mesh_inst = p.add_mesh(uf.output, cmap="jet", clim=[plot_min, self.max_gain], scalar_bar_args=sargs)
-        cad_mesh = self._get_geometry()
-        if cad_mesh:
+        return True
 
-            def toggle_vis_ff(flag):
-                ff_mesh_inst.SetVisibility(flag)
+    @pyaedt_function_handler()
+    def get_far_field_mesh(self, farfield_quantity="RealizedGain", format_quantity="dB10", **kwargs):
+        """Generate a PyVista ``UnstructuredGrid`` object that represents the far field mesh.
 
-            def toggle_vis_cad(flag):
-                for i in cad:
-                    i.SetVisibility(flag)
+        Parameters
+        ----------
+        farfield_quantity : str, optional
+            Far field quantity to plot. The default is ``"RealizedGain"``.
+            Available functions are: ``"RealizedGain"``, ``"RealizedGain_Phi"``, ``"RealizedGain_Theta"``,
+            ``"rEPhi"``, ``"rETheta"``, and ``"rETotal"``.
+        format_quantity : str, optional
+            Conversion data function.
+            Available functions are: ``"abs"``, ``"ang"``, ``"dB10"``, ``"dB20"``, ``"deg"``, ``"imag"``, ``"norm"``,
+            and ``"real"``.
 
-            def scale(value=1):
-                ff_mesh_inst.SetScale(value, value, value)
-                ff_mesh_inst.SetPosition(position)
-                ff_mesh_inst.SetOrientation(rotation_euler)
-                # p.add_mesh(ff_mesh, smooth_shading=True,cmap="jet")
-                return
+        Returns
+        -------
+        :class:`Pyvista.Plotter`
+            ``UnstructuredGrid`` object representing the far field mesh.
 
-            p.add_checkbox_button_widget(toggle_vis_ff, value=True)
-            p.add_text("Show Far Fields", position=(70, 25), color="black", font_size=12)
-            slider_max = int(np.ceil(self.all_max / 2 / self.max_gain))
-            if slider_max > 0:
-                slider_min = 0
-                value = slider_max / 3
-            else:
-                slider_min = slider_max
-                slider_max = 0
-                value = slider_min / 3
-            p.add_slider_widget(scale, [0, slider_max], title="Scale Plot", value=value)
+        """
+        if "convert_to_db" in kwargs:  # pragma: no cover
```

Comment:
I don't like using **kwargs for functions exposed to the users. The functions ends up accepting any argument, without throwing any error. Imagine using `get_far_field_mesh(aaa=123)`,  it would be just fine.